src/components/app/details/appDetails/GraphsModal.tsx

```diff
@@ -86,15 +88,29 @@ export class GraphModal extends Component<GraphModalProps, GraphModalState>{
     }
 
     getNewGraphs(tab: AppMetricsTabType) {
-        let cpu = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'cpu', this.props.newPodHash, this.state.calendarInputs, tab, false);
-        let ram = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'ram', this.props.newPodHash, this.state.calendarInputs, tab, false);
-        let latency = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'latency', this.props.newPodHash, this.state.calendarInputs, tab, false);
-        let status2xx = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'status', this.props.newPodHash, this.state.calendarInputs, tab, false, '2xx');
-        let status4xx = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'status', this.props.newPodHash, this.state.calendarInputs, tab, false, '4xx');
-        let status5xx = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'status', this.props.newPodHash, this.state.calendarInputs, tab, false, '5xx');
-        let status = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'status', this.props.newPodHash, this.state.calendarInputs, tab, false, 'Throughput');
-        let throughput = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'status', this.props.newPodHash, this.state.calendarInputs, tab, false, 'Throughput');
-        let mainChartUrl = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, this.state.mainChartName, this.props.newPodHash, this.state.calendarInputs, tab, true, this.state.statusCode);
+        let k8sVersion = this.props.k8sVersion;
+        let defaultK8sVersion = 'v16.1.10';
+        if (!isK8sVersionValid(k8sVersion)) {
+            k8sVersion = defaultK8sVersion;
+        }
+        let appInfo = {
+            appId: this.props.appId,
+            envId: this.props.envId,
+            environmentName: this.props.environmentName,
+            newPodHash: this.props.newPodHash,
+            k8sVersion: k8sVersion,
+        }
+
+        let cpu = getIframeSrc(appInfo, 'cpu', this.state.calendarInputs, tab, false);
+        let ram = getIframeSrc(appInfo, 'ram', this.state.calendarInputs, tab, false);
+        let latency = getIframeSrc(appInfo, 'latency', this.state.calendarInputs, tab, false);
+        let status2xx = getIframeSrc(appInfo, 'status', this.state.calendarInputs, tab, false, '2xx');
+        let status4xx = getIframeSrc(appInfo, 'status', this.state.calendarInputs, tab, false, '4xx');
+        let status5xx = getIframeSrc(appInfo, 'status', this.state.calendarInputs, tab, false, '5xx');
+        let status = getIframeSrc(appInfo, 'status', this.state.calendarInputs, tab, false, 'Throughput');
```

Comment:
Throughput has T capital, should be avoided, either all small case or all capital

Reply:
Throughput is also a query parameter, where `T` is required to be capital. I have extracted it out in a constant.